frontend/src/horses/HorseOverview.tsx

```diff
@@ -63,7 +64,14 @@ function HorseOverview() {
             .then((response) => response.data)
             .catch((error) => console.error("Error while getting Horses:" + error))
             .then(setHorses)
+    }
 
+    const removeConsumption = (consumptionItemToDelete: ConsumptionModel, editedHorse: HorseModel) => {
+        editedHorse.consumptionList = editedHorse.consumptionList.filter(consumptionItem => consumptionItem !== consumptionItemToDelete)
```

Comment:
I would here pass consumptionItem id instead of whole object and check id in filter
`filter(consumptionItem => consumptionItem.id !== consumptionItemIdToDelete)`.